crates/provider/src/native.rs

```diff
@@ -0,0 +1,323 @@
+use std::error::Error;
+
+use crate::shared::{
+    client::Client,
+    types::{FileMap, NameSpaceDef, PodDef, RunCommandOptions, RunCommandResponse, Settings},
+};
+
+struct Native {
+    // Namespace of the client
+    namespace:             String,
+    // path where configuration relies
+    config_path:           String,
+    // variable that shows if debug is activated
+    debug:                 bool,
+    // the timeout for the client to exit
+    timeout:               u32,
+    // command sent to client
+    command:               String,
+    // temporary directory
+    tmp_dir:               String,
+    pod_monitor_available: bool,
+    local_magic_file_path: String,
+    // name of the provider
+    provider_name:         String,
+}
+
+impl Default for Native {
+    fn default() -> Self {
+        // [TODO]: define the default value for Native
+        todo!()
+    }
+}
+
+impl Native {
+    pub fn new(namespace: &str, config_path: &str, tmp_dir: &str) -> Native {
+        Native {
+            namespace:             namespace.to_owned(),
+            config_path:           config_path.to_owned(),
+            debug:                 true,
+            timeout:               60, // seconds
+            tmp_dir:               tmp_dir.to_owned(),
+            command:               "bash".to_owned(),
+            pod_monitor_available: false,
+            local_magic_file_path: format!("{}/finished.txt", tmp_dir),
+            provider_name:         "native".to_owned(),
+        }
+    }
+}
+
+impl Client for Native {
+    fn create_namespace<'async_trait>() -> core::pin::Pin<
+        Box<
+            dyn core::future::Future<Output = Result<(), Box<dyn Error>>>
+                + core::marker::Send
+                + 'async_trait,
+        >,
+    > {
+        todo!()
+    }
+
+    fn static_setup<'async_trait>(
+        settings: Settings,
+    ) -> core::pin::Pin<
+        Box<
+            dyn core::future::Future<Output = Result<(), Box<dyn Error>>>
+                + core::marker::Send
+                + 'async_trait,
+        >,
+    > {
+        todo!()
+    }
+
+    fn destroy_namespace<'async_trait>() -> core::pin::Pin<
+        Box<
+            dyn core::future::Future<Output = Result<(), Box<dyn Error>>>
+                + core::marker::Send
+                + 'async_trait,
+        >,
+    > {
+        todo!()
+    }
+
+    fn get_node_logs<'async_trait>(
+        pod_name: String,
+        since: Option<u32>,
+        with_timestamp: Option<bool>,
+    ) -> core::pin::Pin<
+        Box<
+            dyn core::future::Future<Output = Result<String, Box<dyn Error>>>
+                + core::marker::Send
+                + 'async_trait,
+        >,
+    > {
+        todo!()
+    }
```

Comment:
```suggestion
    async fn get_node_logs(
        pod_name: String,
        since: Option<u32>,
        with_timestamp: Option<bool>,
    ) -> Result<String, Box<dyn Error>> {
        todo!()
    }
```